x-pack/legacy/plugins/snapshot_restore/common/lib/time_serialization.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { TIME_UNITS } from '../constants';
+
+export const deserializeTime = (time: string) => {
+  const timeUnits = Object.values(TIME_UNITS);
+  const timeChars = time.split('');
+
+  const timeUnit = timeUnits.find(unit => {
+    const unitIndex = timeChars.indexOf(unit);
+    return unitIndex !== -1 && unitIndex === time.length - 1;
+  });
+
+  if (timeUnit) {
+    const timeValue = Number(time.replace(timeUnit, ''));
+
+    if (!isNaN(timeValue)) {
+      return {
+        timeValue,
+        timeUnit,
+      };
+    }
+  }
+
+  return {};
+};
+
+export const serializeTime = (timeValue: number, timeUnit: string) => {
+  return `${timeValue}${timeUnit}`; // e.g., '15d'
```

Comment:
We have intl `formatrelativetime` which might help in this: https://github.com/formatjs/react-intl/blob/master/docs/API.md#formatrelativetime

I am not sure how we'd deserialize it as I am unable to find any tool for that via intl.

Reply:
Thanks for sharing! I was not aware of that.

This is used to support [ES time units](https://www.elastic.co/guide/en/elasticsearch/reference/current/common-options.html#time-units), so I think in this case `formatRelativeTime` doesn't apply.

Reply:
Awesome, that makes sense.